tests/v2/validation/Jenkinsfile_support_matrix_os

```diff
@@ -0,0 +1,238 @@
+
+                  echo awsNodeTemplateParam
+
+                  // RKE2 Node Driver: awsMachineConfig
+
+                  def awsMachineConfigFilename = "awsMachineConfig.yaml"
+                  def awsMachineConfig = env.RANCHER_RKE2_NODE_CONFIG
+
+                  writeFile file: awsMachineConfigFilename, text: awsMachineConfig
+
+                  sh "./yq e '.awsMachineConfig.ami = \"$ami\"' -i awsMachineConfig.yaml"
+
+                  sh "./yq e '.awsMachineConfig.sshUser = \"$sshUser\"' -i awsMachineConfig.yaml"
+
+                  def awsMachineConfigParam = readFile(file: './awsMachineConfig.yaml')
+
+                  echo awsMachineConfigParam
+
+                  // Single custom cluster config: awsEC2Config
+
+                  def awsEC2ConfigFilename = "awsEC2Config.yaml"
+                  def awsEC2Config = env.RANCHER_CUSTOM_CONFIG
+
+                  writeFile file: awsEC2ConfigFilename, text: awsEC2Config 
+
+                  sh "./yq e '.awsEC2Configs.awsEC2Config.[0].awsAMI = \"$ami\"' -i awsEC2Config.yaml"
+
+                  sh "./yq e '.awsEC2Configs.awsEC2Config.[0].awsUser = \"$sshUser\"' -i awsEC2Config.yaml"
+
+                  def awsEC2ConfigParam = readFile(file: './awsEC2Config.yaml')
+
+                  echo awsEC2ConfigParam
+
+                  rke1NodeParams = [
+                    string(name: 'TIMEOUT', value: "${env.TIMEOUT}"),
+                    text(name: 'CONFIG', value: awsNodeTemplateParam),
+                    string(name: 'TEST_PACKAGE', value: "${env.RANCHER_RKE1_TEST_PACKAGE}"),
+                    string(name: 'GOTEST_TESTCASE', value: "${env.RANCHER_RKE1_NODE_GOTEST_TESTCASE}"),
+                  ]
+                  rke2NodeParams = [
+                    string(name: 'TIMEOUT', value: "${env.TIMEOUT}"),
+                    text(name: 'CONFIG', value: awsMachineConfigParam),
+                    string(name: 'TEST_PACKAGE', value: "${env.RANCHER_RKE2_TEST_PACKAGE}"),
+                    string(name: 'GOTEST_TESTCASE', value: "${env.RANCHER_RKE2_NODE_GOTEST_TESTCASE}"),
+                  ]
+                  rke1CustomParams = [
+                    string(name: 'TIMEOUT', value: "${env.TIMEOUT}"),
+                    text(name: 'CONFIG', value: awsEC2ConfigParam),
+                    string(name: 'TEST_PACKAGE', value: "${env.RANCHER_RKE1_TEST_PACKAGE}"),
+                    string(name: 'GOTEST_TESTCASE', value: "${env.RANCHER_RKE1_CUSTOM_GOTEST_TESTCASE}"),
+                  ]
+                  rke2CustomParams = [
+                    string(name: 'TIMEOUT', value: "${env.TIMEOUT}"),
+                    text(name: 'CONFIG', value: awsEC2ConfigParam),
+                    string(name: 'TEST_PACKAGE', value: "${env.RANCHER_RKE2_TEST_PACKAGE}"),
+                    string(name: 'GOTEST_TESTCASE', value: "${env.RANCHER_RKE2_CUSTOM_GOTEST_TESTCASE}"),
+                  ]
+
+                  jobs = [:]
```

Comment:
Maybe I am not able to read this bit; Does this array instantiation happen in the loop?